src/Functions/hasToken.cpp

```diff
@@ -1,26 +1,27 @@
+#include "FunctionFactory.h"
 #include "FunctionsStringSearch.h"
-#include <Functions/FunctionFactory.h>
 #include "HasTokenImpl.h"
-#include <Common/Volnitsky.h>
 
+#include <Common/Volnitsky.h>
 
-namespace DB
-{
 namespace
 {
-
 struct NameHasToken
 {
     static constexpr auto name = "hasToken";
 };
 
-using FunctionHasToken = FunctionsStringSearch<HasTokenImpl<NameHasToken, VolnitskyCaseSensitiveToken, false>>;
-
+using FunctionHasToken = DB::FunctionsStringSearch<DB::HasTokenImpl<NameHasToken, DB::VolnitskyCaseSensitiveToken, false>>;
+using FunctionHasTokenOrNull
+    = DB::FunctionsStringSearch<DB::HasTokenImpl<NameHasToken, DB::VolnitskyCaseSensitiveToken, false>, DB::ExecutionErrorPolicy::Null>;
 }
 
 REGISTER_FUNCTION(HasToken)
 {
-    factory.registerFunction<FunctionHasToken>();
-}
+    factory.registerFunction<FunctionHasToken>(
+        {"Performs lookup of needle in haystack using tokenbf_v1 index."}, DB::FunctionFactory::CaseSensitive);
```

Comment:
(As of now, the official documentation isn't automatically generated from the in-source documentation. It looks like `hasToken()` doesn't have an entry in the official function documentation. But perhaps you could add a new entry to the table here? --> https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#functions-support

Reply:
Sure!